File: moko0428/h_2023_04/카드 뭉치.js
function solution(cards1, cards2, goal) {
  let result = "Yes";

  let c1 = goal.filter((c) => cards1.includes(c));
  let c2 = goal.filter((c) => cards2.includes(c));

  c1.forEach((c, i) => {
    if (c != cards1[i]) result = "No";
  });

  c2.forEach((c, i) => {
    if (c != cards2[i]) answer = "No";
  });
  return "Yes";
}

File: seonghoo1217/src/h_2023_04/programmers/카드_뭉치.java
package h_2023_04.programmers;


public class 카드_뭉치 {
	public String solution(String[] cards1, String[] cards2, String[] goal) {
		String answer = "Yes";
		int card1Idx=0,card1Len=cards1.length;
		int card2Idx=0,card2Len= cards2.length;
		for (int i=0;i<goal.length;i++){
			String point = goal[i];
			if (card1Idx<card1Len&&cards1[card1Idx].equals(point)) card1Idx++;
			else if(card2Idx<card2Len&&cards2[card2Idx].equals(point)) card2Idx++;
			else return "No";
		}
		return answer;
	}
}
